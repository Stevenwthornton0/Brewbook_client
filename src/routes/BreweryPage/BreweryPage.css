.breweryPage {
    padding-left: 40px;
}

.breweryPage > h1 {
    margin: 0;
    padding-top: 20px;
}

.breweryPage > h2 {
    margin-bottom: 0;
}

.contact-container {
    margin-top: 16px;
}

.BreweryPage_reviews-list {
    padding: 0;
    margin-top: 0;
}

.BreweryPage_review {
    display: inline-block;
    border: 1px solid #e0e0e0;
    border-radius: 4px;
    width: 40%;
    padding: 0px 20px;
    margin: 20px 20px 0 0;
    background-color: #e0e0e0;
}

.bold {
    font-weight: 800;
    margin-bottom: 0;
}

.deleteContainer {
    width: 100%;
    margin-bottom: 12px;
    margin-top: 0;
}

.deleteContainer > button {
    width: 30%;
    height: 36px;
    border-radius: 5px;
    box-shadow: 0px 3px 8px #888888;
    font-size: 15px;
    background-color: #fedd02;
    border-color: #fedd02;
    margin-top: 20px; 
  }
  
  button:hover {
    box-shadow: 0px 3px 8px #888888;
    background-color: black;
    border-color: black;
    color: #fedd02;
    cursor: pointer;
  }

  @media screen and (max-width: 500px) {
      .breweryPage {
          width: 85%;
          padding-left: 0;
          margin-left: auto;
          margin-right: auto;
      }

      .BreweryPage_review {
          display: block;
          width: 100%
      }

      .deleteContainer > button {
          width: 30%
      }
  }