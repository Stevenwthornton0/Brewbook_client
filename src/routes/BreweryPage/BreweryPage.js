import React, { Component } from 'react';
import BreweriesApiService from '../../services/breweries-api-service';
import BreweryContext from '../../contexts/BreweryContext';
import ReviewForm from '../../components/ReviewForm/ReviewForm';
import { displayPhone, displayWebsite } from '../../Utils/Utils';
import GoogleMap from '../../components/GoogleMap/GoogleMap';
import './BreweryPage.css';


class BreweryPage extends Component {
    static contextType = BreweryContext;

    constructor(props) {
        super(props)
        this.state = {
            brewery: [],
            reviews: [],
            display: false
        }
    }

    static defaultProps = {
        match: { params: {} }
    }  

    displayMap(latitude, longitude) {
        if (longitude && latitude) {
            return (
                <GoogleMap 
                    latitude={latitude}
                    longitude={longitude}
                /> 
            )
        }
    }

    breweryReviews({ reviews = [] }) {
        return (
            <ul className='BreweryPage_reviews-list'>
                {reviews.map(review => 
                    <li key={review.id} className='BreweryPage_review'>
                        <p>Rating: {review.rating} out of 5</p>
                        <p>"{review.text}"</p>
                        <p>- {review.user.first_name} {review.user.last_name}</p>
                    </li>    
                )}
            </ul>
        )
    }

    componentDidMount() {
        const { breweryId } = this.props.match.params;
        BreweriesApiService.getBreweryById(breweryId)
            .then(this.context.setBrewery)
        BreweriesApiService.getBreweryReviews(breweryId)
            .then(this.context.setReviews)
    }

    componentWillUnmount() {
        this.context.clearBrewery()
    }

    render() {
        const { brewery, reviews, error } = this.context
        let content
        if (error) {
            content = <p>There was an error</p>
        } else if (!brewery.name) {
            content = <p>Loading...</p>
        } else {
            content = 
            <div>
                <h1>{brewery.name}</h1>
                
                {/* {this.displayMap(brewery.latitude, brewery.longitude)} */}
                <div className='contact-container'>
                    {displayPhone(brewery)}
                    {displayWebsite(brewery)}
                </div>
                
                <this.breweryReviews reviews={reviews} />

                <ReviewForm />
            </div>
        }
        return ( 
            <div>
                {content}
            </div> )
    }
}

export default BreweryPage;