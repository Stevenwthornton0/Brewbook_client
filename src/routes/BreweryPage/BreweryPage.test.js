import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import BreweryPage from './BreweryPage';

Enzyme.configure({ adapter: new Adapter() })

describe(`BreweryPage smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = mount(<BreweryPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})