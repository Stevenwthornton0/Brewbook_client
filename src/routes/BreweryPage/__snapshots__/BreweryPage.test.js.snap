// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreweryPage smoke test renders without crashing 1`] = `
<BreweryPage
  match={
    Object {
      "params": Object {},
    }
  }
>
  <div>
    <div
      className="breweryPage"
    >
      <h1 />
      <div
        className="contact-container"
      />
      <h2>
        Reviews
      </h2>
      <breweryReviews
        deleteReview={[Function]}
        reviews={Array []}
      >
        <p>
          No one has posted a review
        </p>
      </breweryReviews>
      <ReviewForm>
        <div>
          <h3>
            Post a Review
          </h3>
          <form
            className="ReviewForm"
            onSubmit={[Function]}
          >
            <label>
              Rating
            </label>
            <select
              className="rating"
              id="rating"
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
            </select>
            <label>
              Review
            </label>
            <Textarea
              aria-label="Leave a review..."
              className="reviewText"
              cols="30"
              id="text"
              placeholder="Leave a review.."
              required={true}
              rows="4"
            >
              <textarea
                aria-label="Leave a review..."
                className="Textarea reviewText"
                cols="30"
                id="text"
                placeholder="Leave a review.."
                required={true}
                rows="4"
              />
            </Textarea>
            <Button
              type="submit"
            >
              <button
                className="Button "
                type="submit"
              >
                Post
              </button>
            </Button>
          </form>
        </div>
      </ReviewForm>
    </div>
  </div>
</BreweryPage>
`;
